spec/services/course/duplication_service_spec.rb

```diff
@@ -38,6 +38,11 @@
           expect(new_course.end_at).to be_within(1.second).of course.end_at + 1.day
         end
 
+        it 'sets new course registration key to nil' do
```

Comment:
Ok, it wasn't saved so no error gets thrown if it doesn't work. I combined this with the basic test for whether a course can be duplicated.